vana/wallet.py

```diff
@@ -227,6 +227,65 @@ def __init__(
         self._coldkeypub = None
         self._mnemonics = {}
 
+        # Create necessary directories
+        self._create_wallet_directories()
+
+        # Check for and handle environment variable based wallet initialization
+        self._init_from_environment()
+
+    def _create_wallet_directories(self):
+        """Create the necessary wallet directory structure if it doesn't exist."""
+        # Create main wallet directory
+        wallet_path = os.path.expanduser(os.path.join(self.path, self.name))
+        # Create hotkeys directory
+        hotkeys_path = os.path.join(wallet_path, "hotkeys")
+
+        # Create directories if they don't exist
+        os.makedirs(hotkeys_path, exist_ok=True)
+
+    def _init_from_environment(self):
+        """
+        Initialize wallet components from environment variables if they exist.
+        Currently supports hotkey initialization from HOTKEY_MNEMONIC.
+        Creates the wallet files on disk without password protection.
+        """
+        hotkey_mnemonic = os.getenv("HOTKEY_MNEMONIC")
```

Comment:
Possible to use wallet args?

https://github.com/vana-com/vana-framework/blob/main/vana/wallet.py#L159